MediaBrowser.Providers/MediaInfo/MediaInfoResolver.cs

```diff
@@ -151,6 +152,48 @@ public abstract class MediaInfoResolver
             return mediaStreams.AsReadOnly();
         }
 
+        /// <summary>
+        /// Retrieves the external streams for the provided audio.
+        /// </summary>
+        /// <param name="audio">The <see cref="Audio"/> object to search external streams for.</param>
+        /// <param name="startIndex">The stream index to start adding external streams at.</param>
+        /// <param name="directoryService">The directory service to search for files.</param>
+        /// <param name="clearCache">True if the directory service cache should be cleared before searching.</param>
+        /// <returns>The external streams located.</returns>
+        public IReadOnlyList<MediaStream> GetExternalStreams(
+            Audio audio,
+            int startIndex,
+            IDirectoryService directoryService,
+            bool clearCache)
+        {
+            if (!audio.IsFileProtocol)
+            {
+                return Array.Empty<MediaStream>();
+            }
+
+            var pathInfos = GetExternalFiles(audio, directoryService, clearCache);
+
+            if (pathInfos.Count == 0)
+            {
+                return Array.Empty<MediaStream>();
+            }
+
+            var mediaStreams = new List<MediaStream>(pathInfos.Count);
+
+            foreach (var pathInfo in pathInfos)
+            {
+                mediaStreams.Add(new MediaStream
+                {
+                    Type = MediaStreamType.Lyric,
+                    Path = pathInfo.Path,
+                    Language = pathInfo.Language,
+                    Index = startIndex++
+                });
+            }
+
+            return mediaStreams.AsReadOnly();
```

Comment:
Normally we don't use `AsReadOnly()` bc every time you use it afterwards everything gets proxied trough `ReadOnlyCollection` which slows everything down